src/main/java/com/kirinpatel/sync/util/DependencyVerifier.java

```diff
@@ -0,0 +1,77 @@
+package com.kirinpatel.sync.util;
+
+import java.io.IOException;
+import java.net.URL;
+
+public class DependencyVerifier {
+
+    private static final boolean IS_64_BIT = System.getProperty("os.arch").contains("64");
+    private static final String OS = System.getProperty("os.name").toLowerCase();
+    private static final String VLC_VERSION = "2.2.8";
+
+    public static void downloadDependencies() {
+        if (OS.contains("win")) {
+            UIMessage.showMessageDialog("The correct version(s) of Java/VLC were unable to be found on\n" +
+                    "your computer. They will be downloaded and ran for you. Please\n" +
+                    "allow them to be run.", "Downloading dependencies");
+            new Thread(() -> {
+                if (!IS_64_BIT){

```

Comment:
Move the check outside of the thread

Reply:
Also, why do you need java on only 32 bit machines.

Reply:
You can't run Java without Java, so they either have 64 or 32 bit, we need them to have 64 bit so we only check for that.

Reply:
I just realized why they were in threads previously. It was to make downloading of multiple files faster, but this is not really necessary.